### PR TITLE
`recursiveCollections`: Remove from all subcollections on delete

### LINKED ISSUE #2686: Remove items from all subcollections when Show Items from Subcollection is enabled
Supersedes #1769 and https://github.com/zotero/zotero/pull/2611

https://forums.zotero.org/discussion/comment/411660/#Comment_411660

I think removing from all collections, with a prompt that says that's what it's doing, is the only thing that makes sense if you have this setting enabled.

### PR DESCRIPTION
**Pull request by AbeJellinek**
Fixes #2686.

### CODE REVIEW

File: chrome/content/zotero/itemTree.jsx [hunk 1]
```diff
@@ -1740,8 +1741,21 @@ var ItemTree = class ItemTree extends LibraryTree {
...
+
 				await Zotero.DB.executeTransaction(async () => {
-					await collectionTreeRow.ref.removeItems(ids);
+					for (let itemID of ids) {
+						for (let collectionID of collectionIDs) {
+							let item = Zotero.Items.get(itemID);
+							item.removeFromCollection(collectionID);
+							await item.save({
+								skipDateModifiedUpdate: true
+							});
```
Comment (dstillman) [inline chrome/content/zotero/itemTree.jsx:1756]: Move `Zotero.Items.get()` and `item.save()` to the outer loop — only need to save once per item

File: chrome/content/zotero/itemTree.jsx [hunk 2]
```diff
@@ -478,7 +478,8 @@ var ItemTree = class ItemTree extends LibraryTree {
 			for (let id of ids) {
 				var split = id.split('-');
 				// Skip if not an item in this collection
-				if (split[0] != collectionTreeRow.ref.id) {
+				if (split[0] != collectionTreeRow.ref.id
+							&& !collectionTreeRow.ref.getDescendents(false, 'collection').some(c => split[0] == c.id)) {
```
Comment (dstillman) [inline chrome/content/zotero/itemTree.jsx:482]: Can skip this if the pref is off, no?

**Review by dstillman — CHANGES_REQUESTED** [review]

File: chrome/content/zotero/itemTree.jsx (hunk 1 above)
Comment (AbeJellinek) [reply chrome/content/zotero/itemTree.jsx:1756]: Right, dumb of me. Thanks.

**Comment by AbeJellinek** [comment]
Should be good now.

**Pull request merged**